### ISSUE TITLE
[Bug]: Narrator is Announcing Nav component as button instead of tab. 

### ISSUE THREAD

**Issue by v-poornima** [microsoft/fluentui#24720]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
  OS: Windows 10 19044.1889
 
  Memory: 146 GB used and 329 GB free
Browsers:
  Edge: Version 105.0.1343.27 (Official build) (64-bit)
```

### Are you reporting Accessibility issue?
yes

### Reproduction

https://codepen.io/poornimak/pen/KKRVyPg

### Bug Description
example link to reproduce issue :https://codepen.io/poornimak/pen/KKRVyPg
Repo steps
1.open the above example 
2. start narrator 
3. narrator announcing the MSN as button and pages as link and Notebook as button.
## Actual Behavior
Narrator Announcing nav Elements as link and buttons.

## Expected Behavior
It should announce it like Notebook tab 

### Logs

```shell
example link to reproduce issue :https://codepen.io/poornimak/pen/KKRVyPg
Repo steps
1.open the above example 
2. start narrator 
3. narrator announcing the MSN as button and pages as link and Notebook as button.
```

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by v-poornima** [comment]
Is There any Update on this?

**Comment by gouttierre** [comment]
Thank you for filing this issue, and i see your repro steps as mentioned. This issue has been queued for upcoming investigations. To set expectations the developers will review this issue once capacity allows and all updates will be posted on the issue accordingly. Note. The developers will be able to confirm if this is a regression, or if this behavior is an issue, or if it was by design at the time.

**Comment by miroslavstastny** [comment]
This is a minimal repro:
```jsx
<Nav
  groups={[
    {
      links: [
        {
          name: 'Rendered as <a>',
          url: 'http://msn.com',
          key: 'key4',
          target: '_blank',
        },
        {
          name: 'Rendered as <button>',
          url: 'http://msn.com',
          key: 'key5',
          disabled: true,
        },
      ],
    },
  ]}
/>
```

All the nav links are rendered as `BaseButton` components.
That component renders an `<a>` (which is narrated as link) if there is an `href` prop (`url` in the `links` object).
But if the link is `disabled` it renders a `<button>` regardless the `href`.
This behavior has been intentionally  introduced in #2127.

**Comment by JustSlone** [comment]
@smhigley can you weigh in on whether this behavior is intentional or not? and it's impact on accessibility?
